packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -167,17 +203,32 @@ export class SelectedNetworkController extends BaseController<
 
     this.update((state) => {
       state.domains[domain] = networkClientId;
-      if (!state.perDomainNetwork) {
-        state.domains[METAMASK_DOMAIN] = networkClientId;
-      }
     });
   }
 
+  #domainHasPermissions(domain: Domain): boolean {
+    return this.messagingSystem.call(
+      'PermissionController:hasPermissions',
+      domain,
+    );
+  }
+
   setNetworkClientIdForDomain(
     domain: Domain,
     networkClientId: NetworkClientId,
   ) {
-    if (!this.state.perDomainNetwork) {
+    // Early return if perDomainNetwork is disabled and the domain is not Metamask, unless it's setting for Metamask specifically
+    if (domain !== METAMASK_DOMAIN && !this.state.perDomainNetwork) {
```

Comment:
These conditions have me a bit confused. What is the purpose of these checks?

I get that we don't want to save a selected network client ID in state for sites without permissions, that makes sense. But what are these other conditions for? Why would it be harmful to save these if `perDomainNetwork` was disabled? What I expected here was `if (!this.#domainHasPermissions(domain)( { return }` or `if (!this.#domainHasPermissions(domain)( { throw new Error(...) }`

Reply:
> Why would it be harmful to save these if perDomainNetwork was disabled

I was thinking it would be harmful because the proxies would then get set to a different networkClient but I'm realizing we don't need to keep the state/proxies synced to the metamask globally selectedNetworkClientId when `perDomainNetwork` is off

Reply:
These conditions have been updated and align more with your thinking here: https://github.com/MetaMask/core/pull/3908/commits/c78443841150f1f0acf3007fa9d67fae9bce5481